STM32/SPI/Src/ADS1120.c

```diff
@@ -150,10 +182,22 @@ static HAL_StatusTypeDef writeRegister(ADS1120Device *dev, const ADS1120_RegConf
     ADS1120Cmd cmd = { .count = (count - 1), .regBegin = offset,  .rwcmd = 4 };
     uint8_t spiReq[1 + 4] = { cmd.byte, 0 };
     memcpy(&spiReq[1], &regs->regs[offset], count);
-    return HAL_SPI_Transmit(dev->hspi, spiReq, 1 + count, ADS_TIMEOUT);
+    if (HAL_SPI_Transmit(dev->hspi, spiReq, 1 + count, ADS_TIMEOUT) != HAL_OK)
+    {
+        return HAL_ERROR;
+    }
+
+    // Check that the registry update was finished correctly.
+    ADS1120_RegConfig regUpdate;
+    memcpy(&regUpdate, &spiReq[1], count);
+    if (verifyTransmission(dev, &regUpdate) != HAL_OK)
```

Comment:
This section will only work correctly if count is 4 and offset is 0. I can see from the rest of the file that this is true, but its a bit of a hack.

Since it is only a single 32-bit register, wouldn't it be cleaner to keep a copy of it locally, update the copy whenever you write to the device, and then compare the copy against the read values?

Reply:
I just changed it such that we check against ADS1120_RegConfig being passed to writeRegister.